### ISSUE TITLE
Hypothesis implementation

### ISSUE THREAD

**Issue by xplosionmind** [wallabag/wallabag#5484]
**Is your feature request related to a problem? Please describe.**
Related to #3839

**Describe the solution you'd like**
Highlights performed in a saved Wallabag page are synchronized and saved to a connected [Hypothesis](https://hypothes.is) account.

**Describe alternatives you've considered**
Highlighting through Hypothesis only, but it is much less handy; otherwise highlighting on Wallabag only, but it is impossible to retrieve and share single highlights, plus it is not possible to view which articles have been highlighted and where.

**Comment by j0k3r** [comment]
I do not think using an external service for highlight is something we want to integrate into wallabag.
The annotation system is part of wallabag and doesn't use an external service for example.
It ease the migration and people which use wallabag tend to want their data on their server instead of somewhere else.
Also, I prefer that idea instead: https://github.com/wallabag/wallabag/issues/5504

**Comment by xplosionmind** [comment]
I understand your point, @j0k3r.  
Nevertheless, I believe that the most appealing feature of open source software is that the community can easily and openly give feedback. I do not believe that arbitrarily closing issues is something that respects this philosophy, nor users and donors who are not developers (otherwise, instead of opening an issue, I would have made a PR).

In any case, allow me to formulate a few more arguments to explain why it is important to consider Hypothesis implementation:
- Hypotesis is open source software and it can be self-hosted.
- Hypothesis is owned by a non profit and data is mostly public with full portability, so self-hosting is not game-changing in this case.
- On top of everything, it is not a matter of whether Hypothesis should be implemented or not, but a matter of **interoperability**, allowing users to **choose** (and respecting their choice) the service they prefer to achieve the result they want. In this specific case, I see Wallabag highlighting features are quite limited and implementing an efficient and renowned service such as Hypothesis would make thing way better!

In conclusion,  
I do not expect that after this comment the feature should be absolutely implemented or gain a greater priority, but at least the issue should not be closed.

**Comment by j0k3r** [comment]
Thanks for that explanation, it would have been better to share your few more arguments first, especially because I didn't know Hypothesis.
It looks like https://annotatorjs.org/ (which we currently use) was based on what Hypothesis has become.

I'll re-open the issue because it might be interesting after all (it's open source, it can be selfhosted, etc.).

What we are thinking about is people will have to create an account or have a self-hosted instance if they want to use annotation (in comparison of the current implementation).

**Comment by xplosionmind** [comment]
Thanks a lot!

<br>

> it would have been better to share your few more arguments first

I will keep this in mind in the future. Thank you very much again.

**Comment by boehs** [comment]
As hypothesis implements WebAnnotation, Isn't this issue moot? Provided I understand the spec correctly, #3839 is the better issue, as then the user could select an annotation endpoint for annotations to be synced to

**Comment by MorganGeek** [comment]
Is there still a plan to implement this ? I'm used to annotate content in Omnivore and I've moved to Wallabag which has better export features however I'm unable to easily find my annotations and highlights and we cannot easily export them.

**Comment by j0k3r** [comment]
Nobody is working on it at the moment.
It's an interesting feature but we are focusing on upgrading Symfony on wallabag for now.
